src/fix_wall_reflect.cpp

```diff
@@ -38,106 +35,110 @@ FixWallReflect::FixWallReflect(LAMMPS *lmp, int narg, char **arg) :
 {
   if (narg < 4) error->all(FLERR,"Illegal fix wall/reflect command");
 
-  dynamic_group_allow = 1;
-
-  // parse args
-
-  nwall = 0;
-  int scaleflag = 1;
-
-  int iarg = 3;
-  while (iarg < narg) {
-    if ((strcmp(arg[iarg],"xlo") == 0) || (strcmp(arg[iarg],"xhi") == 0) ||
-        (strcmp(arg[iarg],"ylo") == 0) || (strcmp(arg[iarg],"yhi") == 0) ||
-        (strcmp(arg[iarg],"zlo") == 0) || (strcmp(arg[iarg],"zhi") == 0)) {
-      if (iarg+2 > narg) error->all(FLERR,"Illegal fix wall/reflect command");
-
-      int newwall;
-      if (strcmp(arg[iarg],"xlo") == 0) newwall = XLO;
-      else if (strcmp(arg[iarg],"xhi") == 0) newwall = XHI;
-      else if (strcmp(arg[iarg],"ylo") == 0) newwall = YLO;
-      else if (strcmp(arg[iarg],"yhi") == 0) newwall = YHI;
-      else if (strcmp(arg[iarg],"zlo") == 0) newwall = ZLO;
-      else if (strcmp(arg[iarg],"zhi") == 0) newwall = ZHI;
-
-      for (int m = 0; (m < nwall) && (m < 6); m++)
-        if (newwall == wallwhich[m])
-          error->all(FLERR,"Wall defined twice in fix wall/reflect command");
-
-      wallwhich[nwall] = newwall;
-      if (strcmp(arg[iarg+1],"EDGE") == 0) {
-        wallstyle[nwall] = EDGE;
-        int dim = wallwhich[nwall] / 2;
-        int side = wallwhich[nwall] % 2;
-        if (side == 0) coord0[nwall] = domain->boxlo[dim];
-        else coord0[nwall] = domain->boxhi[dim];
-      } else if (strstr(arg[iarg+1],"v_") == arg[iarg+1]) {
-        wallstyle[nwall] = VARIABLE;
-        int n = strlen(&arg[iarg+1][2]) + 1;
-        varstr[nwall] = new char[n];
-        strcpy(varstr[nwall],&arg[iarg+1][2]);
-      } else {
-        wallstyle[nwall] = CONSTANT;
-        coord0[nwall] = force->numeric(FLERR,arg[iarg+1]);
-      }
+  if (strcmp(arg[2],"wall/reflect") == 0) { // child class can be stochastic
```

Comment:
Can you invert this line so that the rest of the constructor does not need to 
be indented?  E.g. if the style is stochastic: return

Reply:
> Can you invert this line so that the rest of the constructor does not need to
> be indented? E.g. if the style is stochastic: return

Your suggestion is indeed a simple coding way. My concern is that this stochastic model  is included in the optional package USER-MISC (except if you want to have it in the main directory) and in the future, there may be other wall models in other packages. In that case, the list can be long and is not optimal. The current implementation makes the wall/reflect independent from other wall models and those who do not use those walls, do not have to wonder what they are in the code. 

It is just a question of preference. If you think that including those lines is better, I will change the code.

Reply:
What @sjplimp suggested is indeed the preferred style.